RetailCoder.VBE/Navigation/RegexSearchReplace/RegexSearchReplace.cs

```diff
@@ -119,21 +124,22 @@ private List<RegexSearchResult> SearchCurrentBlock(string searchPattern)
             var state = _parser.State;
             using (var pane = _vbe.ActiveCodePane)
             {
+                if (pane == null || pane.IsWrappingNullReference)
+                {
+                    return new List<RegexSearchResult>();
+                }
+
                 using (var module = pane.CodeModule)
                 {
                     var results = GetResultsFromModule(module, searchPattern);
 
-                    using (var moduleParent = module.Parent)
-                    {
-                        var qualifiedSelection =
-                            new QualifiedSelection(new QualifiedModuleName(moduleParent), pane.Selection);
-                        dynamic block = state.AllDeclarations.FindTarget(qualifiedSelection, declarationTypes).Context
-                            .Parent;
-                        var selection = new Selection(block.Start.Line, block.Start.Column, block.Stop.Line,
-                            block.Stop.Column);
-
-                        return results.Where(r => selection.Contains(r.Selection)).ToList();
-                    }
+                    var qualifiedSelection = pane.GetQualifiedSelection();
+                    dynamic block = state.AllDeclarations.FindTarget(qualifiedSelection.Value, declarationTypes).Context
+                        .Parent;
+                    var selection = new Selection(block.Start.Line, block.Start.Column, block.Stop.Line,
+                        block.Stop.Column);
```

Comment:
This is beyond PR's scope but just to ask that this might need to use `EndLine` and `EndColumn` instead?